ads/imedia.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {loadScript, computeInMasterFrame, validateData} from '../3p/3p';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function imedia(global, data) {
+  validateData(data, ['id', 'positions']);
+
+  let positions = null;
+  if (data.positions) {
+    positions = JSON.parse(data.positions);
+  }
+  const mW = context.isMaster ? global : context.master;
+
+  // create parent element
+  const parentElement = document.createElement('div');
+  parentElement.id = data.id;
+  global.document.getElementById('c').appendChild(parentElement);
+
+  // array of all ad elements through all iframes
+  if (!mW.elements) {
+    mW.elements = [];
+  }
+  mW.elements.push(parentElement);
+
+  computeInMasterFrame(global, 'imedia-load', function(done) {
```

Comment:
If I do:
```
computeInMasterFrame(global, 'imedia-load', done => {
```
Then gulp check-types raise:

```
Error compiling ./3p/integration.js
Command failed: java -jar -XX:+TieredCompilation build-system/runner/dist/runner.jar --flagfile="/var/folders/d0/h4nrdlvn4l7dmcs4534jc5rsm20kq1/T/13a18c8d-7bde-48df-82ce-294ce769fb0e" --compilation_level=SIMPLE_OPTIMIZATIONS --assume_function_wrapper=true --language_in=ECMASCRIPT6 --language_out=ECMASCRIPT5 --rewrite_polyfills=false --externs=build-system/amp.extern.js --externs=third_party/closure-compiler/externs/intersection_observer.js --externs=third_party/closure-compiler/externs/shadow_dom.js --externs=third_party/closure-compiler/externs/web_animations.js --externs=ads/ads.extern.js --externs=build-system/amp.oti.extern.js --js_module_root=node_modules/ --js_module_root=build/patched-module/ --js_module_root=build/fake-module/ --entry_point=./3p/integration.js --process_common_js_modules=true --only_closure_dependencies=true --output_wrapper=(function(){%output%})();
//# sourceMappingURL=integration-check-types.js.map
 --create_source_map=build/cc/_3p_integration.js.map --source_map_location_mapping=|http://localhost:8000/ --warning_level=DEFAULT --jscomp_off=unknownDefines --define=TYPECHECK_ONLY=true --hide_warnings_for=third_party/caja/ --hide_warnings_for=third_party/closure-library/sha384-generated.js --hide_warnings_for=third_party/d3/ --hide_warnings_for=third_party/vega/ --hide_warnings_for=third_party/webcomponentsjs/ --hide_warnings_for=node_modules/ --hide_warnings_for=build/patched-module/ --hide_warnings_for=3p/environment.js --jscomp_error=checkTypes --jscomp_error=accessControls --jscomp_error=const --jscomp_error=constantProperty --jscomp_error=globalThis --conformance_configs=build-system/conformance-config.textproto --js_output_file="build/cc/_3p_integration.js"
ads/imedia.js:41: ERROR - dangerous use of the global this object
      if (this.im != null) {
          ^

ads/imedia.js:42: ERROR - dangerous use of the global this object
        mW.im = this.im;
                ^

2 error(s), 0 warning(s)
```

so I use the same code like in kargo.js with function instead of arrow function ( check-types passed):

```
computeInMasterFrame(global, 'imedia-load', function (done) {
```

Or is it better way to do:
```
/* global im: false */
```

and then to have an arrow function (and _im_ without _this_ inside):

```
computeInMasterFrame(global, 'imedia-load', done => {
  loadScript(this, 'https://i.imedia.cz/js/im3.js', () => {
    if (im != null) {
      mW.im = im;
```